hazelcast/src/main/java/com/hazelcast/cache/impl/merge/policy/CacheMergePolicyProvider.java

```diff
@@ -64,10 +68,14 @@ private void addOutOfBoxPolicies() {
         }
     }
 
-    public CacheMergePolicy getMergePolicy(String className) {
+    public Object getMergePolicy(String className) {
```

Comment:
can we add an explanatory javaDoc here about what exactly this method does? Also can we return an exact type instead of `Object` to prevent possible errors?

Reply:
I'll add some JavaDoc, but there is no shared interface between the old and new merge policies, so there is no better type than `Object`. I thought of leaving the legacy providers as they are, but then the logic in `SplitBrainHandlerService` becomes more complex (and differs even more from the other new implementations).